### PR TITLE
refactor: claim.ts and refund.ts

### PR DESCRIPTION
**Pull request by dni**
moved from helper.ts

### CODE REVIEW

File: tests/components/RefundButton.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        expect(button.disabled).not.toBeTruthy();
+
+        // paste a invalid address
+        fireEvent.input(input, {
+            target: {
+                value: "XXXQ5FhU2497BryFfUgbqkAJE87aKHUhXMp",
+            },
+        });
+
+        expect(button.disabled).toBeTruthy();
```
Comment (michael1011) [inline tests/components/RefundButton.spec.tsx:68]: That should be a separate test case

File: src/components/RefundButton.tsx [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    setRefundAddress,
+    setRefundTx,
+} from "../signals";
+import { getAddress, getNetwork } from "../utils/compat";
+import { decodeInvoice } from "../utils/invoice";
+import { refund } from "../utils/refund";
+import { prefix0x, satoshiToWei } from "../utils/rootstock";
+import ContractTransaction from "./ContractTransaction";
+
+const RefundButton = ({ swap }: { swap: Accessor<Record<string, any>> }) => {
```
Comment (michael1011) [inline src/components/RefundButton.tsx:21]: The name is confusing
It's but just a button

File: src/components/SwapChecker.tsx [hunk 3]
```diff
@@ -60,25 +61,30 @@ export const SwapChecker = () => {
...
+        const currentSwap = swaps().find((s) => activeSwap.id === s.id);
         if (
             currentSwap.claimTx === undefined &&
             data.transaction !== undefined &&
             (data.status === swapStatusPending.TransactionConfirmed ||
                 data.status === swapStatusPending.TransactionMempool)
         ) {
-            claim(currentSwap, data.transaction);
+            claim(currentSwap, data.transaction, (swap) => {
+                // TODO: this get the swap status updated with the claim tx
```
Comment (michael1011) [inline src/components/SwapChecker.tsx:84]: ?

File: src/utils/claim.ts [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+        true,
+        assetHash,
+        blindingKey,
+    );
+};
+
+export const claim = async (
+    swap: any,
+    swapStatusTransaction: { hex: string },
+    callback: (swap: any) => any,
```
Comment (michael1011) [inline src/utils/claim.ts:48]: We should get rid of the callbacks. ***Especially*** in async functions

**Review by michael1011 — CHANGES_REQUESTED** [review]

File: src/utils/claim.ts (hunk 4 above)
Comment (dni) [reply src/utils/claim.ts:48]: agreed, next pr

File: tests/components/RefundButton.spec.tsx (hunk 1 above)
Comment (dni) [reply tests/components/RefundButton.spec.tsx:89]: done

File: src/components/SwapChecker.tsx (hunk 3 above)
Comment (dni) [reply src/components/SwapChecker.tsx]: done

File: tests/components/RefundButton.spec.tsx [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+            },
+        });
+
+        const button = (await screen.findByTestId(
+            "refundButton",
+        )) as HTMLButtonElement;
+        expect(button.disabled).not.toBeTruthy();
+    });
+
+    test("button should be inactive after pasting in address", async () => {
```
Comment (michael1011) [inline tests/components/RefundButton.spec.tsx:62]:
```suggestion
    test("button should be inactive after pasting an invalid address", async () => {
```

**Review by michael1011 — APPROVED** [review]

**Pull request merged**